KabuKit/Classes/core/Swift/SceneSequence.swift

```diff
@@ -46,32 +53,102 @@ public class SceneSequence<ContextType, GuideType: Guide> : Scene, SceneContaine
         completion()
     }
     
+    public func contain<SceneType: Scene>(_ scene: SceneType) -> Bool {
+        return self.scenes.contains { (screen) -> Bool in
+            (scene === screen)
+        }
+    }
+   
     /**
-     Sequenceをスタートさせる
+     Sequenceをスタートさせます
+     
+     - Attention:
+     起動時の処理のみ規定するため、leaveが呼ばれてもこのsequenceを停止させる処理は動かず
+     
+     このSequenceに追加された一番最初のSceneはreleaseされません
      
      - Parameters:
-       - stage: Sceneが乗っかるstageオブジェクト
-       - scene: 起動させた際に一番最初に表示させる画面
-       - context: sceneを表示させるさいに必要となるcontextオブジェクト
-       - invoke: sceneを表示させるのに必要となる処理
+       - stage: 画面遷移の元の `stage` となるインスタンス
+       - setup: Sequenceをスタートさせる際に画面表示など必要となるメソッド
      */
-    public func startWith<SceneType: Scene>(_ stage: StageType,
-                                    _ scene: SceneType,
-                                    _ context: SceneType.Context,
-                                    _ invoke: @escaping (_ scene: SceneType, _ stage: StageType) -> Void) {
+    public func start(on stage: StageType,
+                      with setup: @escaping (_ scene: FirstScene, _ stage: StageType) -> Void) {
+        self.start(on: stage, with: setup, {})
+    }
+    
+    
+    /**
+     Sequenceをスタートさせます
+     
+     - Parameters:
+       - stage: 画面遷移の元の `stage` となるインスタンス
+       - setup: Sequenceをスタートさせる際に画面表示など必要となるメソッド
+       - completion: Sequenceのsetupが完了したら呼ばれます
+     */
+    public func start(on stage: StageType,
+                      with setup: @escaping (_ scene: FirstScene, _ stage: StageType) -> Void,
+                      _ completion: @escaping () -> Void  = {}) {
         let operation = SceneOperation<StageType>()
         self.stage = stage
         self.guide?.start(with: operation)
+        // TODO FIX, Any Pattern
+        guard let scenario = operation.resolve(from: firstScene) else { return }
+        scenario.setup(at: firstScene, on: stage, with: self, when: nil) {
+            let hashwrap = ScreenHashWrapper(self.firstScene)
+            self.scenes.append(self.firstScene)
+            procedureByScene[hashwrap] = scenario
+            contextByScreen[hashwrap] = self.context
+            setup(self.firstScene, stage)
+            completion()
+            self.isStarted = true
+        }
+    }
 
+    public func start(on stage: StageType,
+                      accordingAs transitioningRule: @escaping (_ scene: FirstScene, _ stage: StageType) -> (_ screens: [Screen]) -> Void) {
```

Comment:
I think that to use `transitioningRule` (or `rule`) for argument label will make more clear.